src/barometer.cpp

```diff
@@ -0,0 +1,47 @@
+#include "barometer.hpp"
+
+#include <Arduino.h>
+
+namespace obc {
+void init(BMP &bmp)
+{
+    Serial.println("Initializing BMP module");
+
+    if (bmp.begin() == 0) {
+        Serial.println(
+            "BMP init failed. Please check connections "
+            "or reboot");
+        while (true) {}
+    }
+    else
+        Serial.println("BMP init success!\n\n");
+
+    bmp.setOversampling(4);
+}
+
+bmpMeasurements bmp_measurements(BMP &bmp)
+{
+    char result = bmp.startMeasurment();
+    Measurements temp = {0, 0, 0};
+
+    if (result == 0) { return {{}, Error::Busy}; }
+
+    result = bmp.getTemperatureAndPressure(temp.temperature, temp.pressure);
+
+    if (result == 0) { return {{}, Error::Busy}; }
+
+    temp.altitude = bmp.altitude(temp.pressure, a_pr);
+
+    return {{temp.temperature, temp.pressure, temp.altitude}, Error::Ok};
+}
```

Comment:
Alexander Stepanov zwykł mawiać zeby nie odrzucać żadnych informacji. Obecnie jak wywali sie tylko altitude() to tracimy wszystko.

W Resulcie mozemy zwracac dodatkowe informacje odnośnie przyczyny błędu i zwrócić przynajmniej część danych którą udało się pobrać.

Albo rozdzielić te funkcje na 3 różne z osobnymi Resultami. Ale to mi wygląda na troche marnowanie pamięci +  wtedy równie dobrze można używać bezposrednio obiektu BMP280. Przydałby sie jakis inny sposób na oznaczenie co jest ok a co nie.

Ogolnie nie ma to dla nas większego znaczenia w tym momencie, taka bardziej dygresja.

Reply:
Tutaj sugerowałem się biblioteką do BMP, bo jeśli nie mamy pierwszego pomiaru czyli w tym wypadku pomiarów z funkcji getTemperatureAndPressure to nie możemy mieć pomiaru altitude.

Dlatego myślę, że returny mogą pozostać tak jak są, bo jeśli nie będziemy mieli pomiaru temperatury i ciśnienia to po co nam wysokość która będzie zwracała krzaczki. 

Można dodać jeszcze informację czy wysypało się na pierwszym pomiarze czy na drugim.